k8s/upgrade/src/bin/upgrade-job/upgrade.rs

```diff
@@ -132,3 +164,30 @@ async fn upgrade_product(opts: &CliArgs, event: &mut EventRecorder) -> Result<()
 
     Ok(())
 }
+
+fn partial_rebuild_check(
+    yet_to_upgrade_io_engine_pods: &[Pod],
+    partial_rebuild_is_enabled: bool,
+) -> Result<()> {
+    let partial_rebuild_disable_required = yet_to_upgrade_io_engine_pods
+        .iter()
+        .filter_map(|pod| pod.labels().get(CHART_VERSION_LABEL_KEY))
+        .any(|v| {
+            let version =
+                Version::parse(v).expect("failed to parse version from io-engine Pod label");
+            version.ge(&PARTIAL_REBUILD_DISABLE_EXTENTS.0)
+                & version.le(&PARTIAL_REBUILD_DISABLE_EXTENTS.1)
+        });
+
+    if partial_rebuild_disable_required && partial_rebuild_is_enabled {
+        error!("Partial rebuild must be disabled for upgrades from {CORE_CHART_NAME} chart versions >= {}, <= {}", PARTIAL_REBUILD_DISABLE_EXTENTS.0, PARTIAL_REBUILD_DISABLE_EXTENTS.1);
```

Comment:
@tiagolobocastro @Abhinandan-Purkait @tiagolobocastro -- does this error message seem ok to you?

Reply:
LGTM. Maybe can we specify the exact value as well if that adds to a better UX?

Reply:
Two different charts to cover, do I do it for both?

Reply:
We can just do the mayastor specific version, then the user can always lookup the docs since they will have to anyway to figure out how to disable partial rebuild?

Reply:
I'm not adding the helm command here, as this plugin doesn't support the modification via itself. The helm command for this change should be picked up from the docs IMO.